integration-tests/bats/system-tables.bats

```diff
@@ -531,6 +531,7 @@ SQL
 @test "system-tables: cannot delete last branch in dolt_branches" {
```

Comment:
I would call this test "system-tables: dolt_branches is read-only".

Then I would add update and insert statement tests.

Then anywhere else there was a write to this table, just delete that part of the test and replace it with the equivalent `call dolt_branch()` call.

This makes the read-only test all in one place instead of spread out all over the code.